src/migmose/mig/reducednestednachrichtenstruktur.py

```diff
@@ -2,16 +2,135 @@
 contains class for trees consisting of segments of mig tables
 """
 
-from typing import Optional
+import json
+from pathlib import Path
+from typing import Any, Optional
 
+from loguru import logger
+from maus.edifact import EdifactFormat
 from pydantic import BaseModel
 
 from migmose.mig.nachrichtenstrukturzeile import NachrichtenstrukturZeile
+from migmose.mig.nestednachrichtenstruktur import NestedNachrichtenstruktur
 
 
 class ReducedNestedNachrichtenstruktur(BaseModel):
     """will contain the tree structure of nachrichtenstruktur tables"""
 
     header_linie: Optional[NachrichtenstrukturZeile] = None
-    segmente: list[NachrichtenstrukturZeile]
-    segmentgruppe: list["ReducedNestedNachrichtenstruktur"]
+    segmente: list[Optional[NachrichtenstrukturZeile]] = []
+    segmentgruppen: list[Optional["ReducedNestedNachrichtenstruktur"]] = []
+
+    @classmethod
+    def create_reduced_nested_nachrichtenstruktur(
+        cls, nachrichten_struktur: NestedNachrichtenstruktur
+    ) -> "ReducedNestedNachrichtenstruktur":
+        """init nested Nachrichtenstruktur"""
+
+        # Helper function to create a unique identifier for each segment
+        def get_identifier(segment: NachrichtenstrukturZeile | None) -> tuple[str, str]:
+            if segment is None:
+                return ("0", "root")
+            return (segment.zaehler, segment.bezeichnung)
+
+        def count_segments(segment_group: ReducedNestedNachrichtenstruktur | NestedNachrichtenstruktur) -> int:
+            # Start with counting segments directly under the current segment group
+            total_segments = len(segment_group.segmente)
+            # Recursively count segments in nested segment groups
+            for nested_sg in segment_group.segmentgruppen:
+                if nested_sg is not None:
+                    total_segments += count_segments(nested_sg)
+
+            return total_segments
+
+        # Function to process segments and remove duplicates within the same list.
+        def process_segments(segments: list[Optional[NachrichtenstrukturZeile]]):
+            seen = set()
```

Comment:
just to let you know.
You can also use the following to create an empty set
```python
seen = { }
```
The space is important, it would be an empty dictionary instead.